allennlp/training/metrics/fbeta_multi_label_measure.py

```diff
@@ -0,0 +1,210 @@
+        self._threshold = threshold
+
+        # statistics
+        # the total number of true positive instances under each class
+        # Shape: (num_classes, )
+        self._true_positive_sum: Union[None, torch.Tensor] = None
+        # the total number of instances
+        # Shape: (num_classes, )
+        self._total_sum: Union[None, torch.Tensor] = None
+        # the total number of instances under each _predicted_ class,
+        # including true positives and false positives
+        # Shape: (num_classes, )
+        self._pred_sum: Union[None, torch.Tensor] = None
+        # the total number of instances under each _true_ class,
+        # including true positives and false negatives
+        # Shape: (num_classes, )
+        self._true_sum: Union[None, torch.Tensor] = None
+
+    @overrides
+    def __call__(
+        self,
+        predictions: torch.Tensor,
+        gold_labels: torch.Tensor,
+        mask: Optional[torch.BoolTensor] = None,
+    ):
+        """
+        # Parameters
+
+        predictions : `torch.Tensor`, required.
+            A tensor of predictions of shape (batch_size, ..., num_classes).
+        gold_labels : `torch.Tensor`, required.
+            A tensor of integer class label of shape (batch_size, ...). It must be the same
+            shape as the `predictions` tensor without the `num_classes` dimension.
+        mask : `torch.BoolTensor`, optional (default = `None`).
+            A masking tensor the same size as `gold_labels`.
+        """
+        predictions, gold_labels, mask = self.detach_tensors(predictions, gold_labels, mask)
+        device = gold_labels.device
+
+        # Calculate true_positive_sum, true_negative_sum, pred_sum, true_sum
+        num_classes = predictions.size(-1)
+        if (gold_labels >= num_classes).any():
+            raise ConfigurationError(
+                "A gold label passed to FBetaMeasure contains "
+                f"an id >= {num_classes}, the number of classes."
+            )
+
+        # It means we call this metric at the first time
+        # when `self._true_positive_sum` is None.
+        if self._true_positive_sum is None:
+            self._true_positive_sum = torch.zeros(num_classes, device=predictions.device)
+            self._true_sum = torch.zeros(num_classes, device=predictions.device)
+            self._pred_sum = torch.zeros(num_classes, device=predictions.device)
+            self._total_sum = torch.zeros(num_classes, device=predictions.device)
+
+        if mask is None:
+            mask = torch.ones_like(gold_labels).bool()
+        gold_labels = gold_labels.float()
+
+        # If the prediction tensor is all zeros, the record is not classified to any of the labels.
+        pred_mask = (predictions.sum(dim=-1) != 0).unsqueeze(-1)
+        threshold_predictions = torch.where(
+            predictions >= self._threshold,
+            torch.ones_like(predictions),
+            torch.zeros_like(predictions),
+        )
```

Comment:
Isn't that the same as
```suggestion
        threshold_predictions = (predictions >= self._threshold).float()
```

Reply:
It is! I've replace the code with your suggestion 👍